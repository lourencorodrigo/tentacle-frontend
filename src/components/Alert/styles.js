import styled, { keyframes, css } from 'styled-components';
import { darken } from 'polished';

import { em } from '../../styles/tools';

const slideInDown = keyframes`
  0%,
  60%,
  75%,
  90%,
  to {
    animation-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);
  }
  0% {
    opacity: 0;
    transform: translate3d(0, -3000px, 0);
  }
  60% {
    opacity: 1;
    transform: translate3d(0, 25px, 0);
  }
  75% {
    transform: translate3d(0, -10px, 0);
  }
  90% {
    transform: translate3d(0, 5px, 0);
  }
  to {
    transform: none;
  }
`;

export const Wrapper = styled.div`
  cursor: pointer;
  position: absolute;
  left: 0;
  right: 0;
  border-bottom: solid ${em(1)};
  box-shadow: rgba(0, 0, 0, 0.16) 0px 3px 4px;
  display: flex;
  justify-content: center;
  padding: ${em(10)};
  animation: ${slideInDown} 1s linear;
  opacity: 0.9;

  ${props =>
    props.color === 'success' &&
    css`
      background-color: ${props => props.theme.alert_success};
      border-color: ${darken(0.1, props.theme.alert_success)};
    `};

  ${props =>
    props.color === 'info' &&
    css`
      background-color: ${props => props.theme.alert_info};
      border-color: ${darken(0.1, props.theme.alert_info)};
    `};

  ${props =>
    props.color === 'warning' &&
    css`
      background-color: ${props => props.theme.alert_warning};
      border-color: ${darken(0.1, props.theme.alert_warning)};
    `};

  ${props =>
    props.color === 'danger' &&
    css`
      background-color: ${props => props.theme.alert_danger};
      border-color: ${darken(0.1, props.theme.alert_danger)};
    `};
`;

export const Title = styled.h1`
  margin: 0;
  font-family: Avenir;
  font-size: ${em(16)};

  ${props =>
    props.color === 'success' &&
    css`
      color: ${darken(0.6, props.theme.alert_success)};
    `};

  ${props =>
    props.color === 'info' &&
    css`
      color: ${darken(0.6, props.theme.alert_info)};
    `};

  ${props =>
    props.color === 'warning' &&
    css`
      color: ${darken(0.6, props.theme.alert_warning)};
    `};

  ${props =>
    props.color === 'danger' &&
    css`
      color: ${darken(0.6, props.theme.alert_danger)};
    `};
`;

export const Container = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Button = styled.button`
  background-color: transparent;
  border: 0;
  cursor: pointer;
  color: inherit;
  font-size: ${em(20)};
  font-weight: 400;
  line-height: 1;
  color: rgba(43, 43, 43, 0.8);
  text-shadow: 0 1px 0 #fff;

  &:focus {
    outline: 0;
  }
`;

export const Span = styled.span`
  font-size: ${em(24)};
  font-weight: 700;
  line-height: 1;
  text-shadow: 0 ${em(1)} 0 #fff;
  opacity: 0.5;
  transition: opacity 0.3s linear;

  &:focus,
  &:hover {
    color: #000;
    text-decoration: none;
    opacity: 0.75;
  }
`;
