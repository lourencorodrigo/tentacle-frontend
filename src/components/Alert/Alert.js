import React from 'react';
import PropTypes from 'prop-types';

import { Wrapper, Title, Container } from './styles';

const Alert = ({ color, onClick, children }) => (
  <>
    {true && (
      <Wrapper color={color} onClick={onClick}>
        <Container>
          <Title color={color}>{children}</Title>
        </Container>
      </Wrapper>
    )}
  </>
);

Alert.defaultProps = {
  color: 'success'
};

Alert.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string.isRequired,
    PropTypes.element.isRequired
  ]),
  color: PropTypes.oneOf(['success', 'info', 'danger', 'warning']),
  onClick: PropTypes.func
};

export default Alert;
