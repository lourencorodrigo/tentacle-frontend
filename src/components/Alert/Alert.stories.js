import React from 'react';
import { storiesOf } from '@storybook/react';

import Alert from './Alert';

storiesOf('Alert', module).add('success', () => (
  <Alert type="success">Test Success</Alert>
));

storiesOf('Alert', module).add('info', () => (
  <Alert type="info">Test Info</Alert>
));

storiesOf('Alert', module).add('warning', () => (
  <Alert type="warning">Test Warning</Alert>
));

storiesOf('Alert', module).add('danger', () => (
  <Alert type="danger">Test Danger</Alert>
));
