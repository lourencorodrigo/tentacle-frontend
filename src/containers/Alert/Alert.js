/* eslint-disable */
import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import Alert from '../../components/Alert';
import { ALERT_SERVICE_WORKER } from '../../utils/constants';

const reloadPage = () => {
  window.location.reload(true);
};

const selectAlert = (title, id) => {
  switch (id) {
    case ALERT_SERVICE_WORKER:
      return (
        <Alert isClickFull={true} onClick={reloadPage} color="success">
          <FormattedMessage id={title} />
        </Alert>
      );
    default:
      return <Alert color="danger">{title}</Alert>;
  }
};

const renderAlert = (show, title, id) => (show ? selectAlert(title, id) : null);

const AlertContainer = ({ alertState }) =>
  renderAlert(...Object.values(alertState));

AlertContainer.propTypes = {
  alertState: PropTypes.object
};

export default AlertContainer;
