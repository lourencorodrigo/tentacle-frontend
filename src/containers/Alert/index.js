import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Alert from './Alert';
import * as AlertAction from '../../actions/alert';

const mapStateToProps = state => ({
  alertState: state.alert
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(AlertAction, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Alert);
