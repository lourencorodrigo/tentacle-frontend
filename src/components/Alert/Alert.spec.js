import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestRenderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Alert from './';

describe('alert component', () => {
  let wrapper;
  const alert = (
    <Alert type="success">
      <span id="test">Test</span>
    </Alert>
  );

  beforeEach(() => {
    wrapper = mount(alert);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(alert, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should compare the text.', () => {
    expect(
      wrapper
        .find('#test')
        .render()
        .text()
    ).toEqual('Test');
  });

  it('should compare the component with a snapshot', () => {
    const component = ReactTestRenderer.create(alert);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
