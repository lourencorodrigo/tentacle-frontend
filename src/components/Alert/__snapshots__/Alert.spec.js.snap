// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert component should compare the component with a snapshot 1`] = `
<div
  className="sc-bdVaJa ewZunz"
  type="success"
>
  <span
    id="test"
  >
    Test
  </span>
  <button
    aria-label="Close"
    className="sc-bwzfXH iEoEWc"
    data-dismiss="alert"
    type="button"
  >
    <span
      aria-hidden="true"
      className="sc-htpNat kIqkdN"
    >
      Ã—
    </span>
  </button>
</div>
`;
